### PR TITLE
538 status changes to received

### PR DESCRIPTION
**Pull request by marcellmueller**
This is for issue #538. 

Definitely open to more ideas for test cases. I know @matthieu-foucault said we should ideally use the `create_application` function in tests though since I needed to add specific intakes to them I stuck with `insert`, let me know if you have other thoughts thanks.

### CODE REVIEW

File: db/deploy/create_roles.sql [hunk 1]
```diff
@@ -10,7 +10,7 @@ begin
     select true
     from pg_catalog.pg_roles
     where rolname = 'ccbc_guest') then
-    
+
```
Comment (AntBush) [inline db/deploy/create_roles.sql:13]: It looks like we did a rework and then didn't change the file other than some spacing fixes (I'm guessing be prettier). I think we should remove the rework for readability

File: db/deploy/functions/receive_applications.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    open applications for select id, intake_id
+		    from ccbc_public.application
+		    where intake_id in (
+                select id from ccbc_public.intake where now() >= close_timestamp);
+
+    loop
+        fetch applications into current_app;
+        exit when not found;
+
+        select s.status into last_status from ccbc_public.application_status s inner join
```
Comment (AntBush) [inline db/deploy/functions/receive_applications.sql:27]: We can probably use the `application_status` computed column function here to simplify this, we should add execute permission for that function. There's also the possibility that how we store an application status changes and then we'd have two places to have to check/change.

File: db/revert/ccbc_job_executor_permissions.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+-- Revert ccbc:ccbc_job_executor_permissions from pg
+
+begin;
+
+-- XXX Add DDLs here.
```
Comment (AntBush) [inline db/revert/ccbc_job_executor_permissions.sql:5]: I'm guessing we want a revert in here if we want to remove this role at one point? Looking at the old revert for `create_roles` it implies we only want to do it manually so I'm not sure, maybe @matthieu-foucault can shed some light on this

File: db/test/unit/functions/receive_applications_test.sql [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+delete from ccbc_public.application;
+delete from ccbc_public.application_status;
+
+insert into
+  ccbc_public.intake(id, open_timestamp, close_timestamp, ccbc_intake_number)
+overriding system value
+values
+  (1, '2022-08-19 09:00:00 America/Vancouver','2022-11-06 09:00:00 America/Vancouver', 1),
+  (2, '2021-08-19 09:00:00 America/Vancouver','2021-11-06 09:00:00 America/Vancouver', 2);
+
```
Comment (AntBush) [inline db/test/unit/functions/receive_applications_test.sql:22]: We should add the `mocks.set_mocked_time_in_transaction` function here as the tests will start to fail after Nov 6

File: db/test/unit/roles/create_roles_test.sql [hunk 5]
```diff
@@ -14,6 +14,11 @@ select isnt_superuser(
     'ccbc_guest should not be a super user'
 );
 
+select has_role( 'ccbc_guest', 'role ccbc_job_executor exists' );
```
Comment (AntBush) [inline db/test/unit/roles/create_roles_test.sql:17]:
```suggestion
select has_role( 'ccbc_job_executor', 'role ccbc_job_executor exists' );
```

File: db/verify/functions/receive_applications.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
+-- Verify ccbc:functions/receive_applications on pg
+
+BEGIN;
+
```
Comment (AntBush) [inline db/verify/functions/receive_applications.sql:4]: We can just delete verify files as we aren't using them anymore, same applies to the rest of the `verify` files

**Review by AntBush — CHANGES_REQUESTED** [review]
Appreciate the work! It's good stuff, just a couple minor tweaks

File: db/deploy/create_roles.sql (hunk 1 above)
Comment (marcellmueller) [reply db/deploy/create_roles.sql:13]: I'm confused as to what you mean, we did the sqitch rework to add the `ccbc_job_executor` in `create_roles@1.5.0.sql`. This white space change is from the pre commit hook.

Comment (AntBush) [reply db/deploy/create_roles.sql:13]: Ah I misunderstood, the `.sql` file without the tag is the latest one while the tagged version is the old version, so you have them swapped in this case. Just need to change the file names in this case then. When this is deployed in prod it won't create the latest roles

File: db/deploy/functions/receive_applications.sql (hunk 2 above)
Comment (marcellmueller) [reply db/deploy/functions/receive_applications.sql:27]: @VB-at-Bis thoughts on this?

File: app/cypress/integration/upload.spec.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
+/// <reference types="Cypress" />
+
+context('Homepage', () => {
```
Comment (AntBush) [inline app/cypress/integration/upload.spec.js:4]: I think this got added in by accident

File: db/revert/ccbc_job_executor_permissions.sql (hunk 3 above)
Comment (marcellmueller) [reply db/revert/ccbc_job_executor_permissions.sql:5]: @AntBush okay I took care of everything except for this, take a look when you get time. 

@matthieu-foucault did you have any thoughts on this comment?

Comment (matthieu-foucault) [reply db/revert/ccbc_job_executor_permissions.sql:5]: > thoughts

The comment in the old revert from `create_roles` don't necessarily apply to this project.

Comment (marcellmueller) [reply db/revert/ccbc_job_executor_permissions.sql:5]: I'm not sure I get the point of this, maybe if we dropped the other roles on revert but given how `create_roles` revert currently is I'm not going to bother.

**Review by AntBush — APPROVED** [review]

**Pull request merged**